onnxruntime/core/framework/utils.cc

```diff
@@ -568,5 +568,42 @@ int32_t ONNXTensorElementDataTypeToProtoTensorType(ONNXTensorElementDataType onn
   }
 }
 
+common::Status VerifyTensorsAllocatedContiguously(OpKernelContext* context) {
+  const Tensor* prev_input = context->Input<Tensor>(0);
+  Tensor* prev_output = context->Output(0, prev_input->Shape());
+  prev_output->SetByteOffset(prev_input->ByteOffset());
+  for (int i = 1; i < context->InputCount(); i++) {
+    const Tensor* curr_input = context->Input<Tensor>(i);
+    Tensor* curr_output = context->Output(i, curr_input->Shape());
```

Comment:
Hmmm ... strictly speaking the Kernel is allocating its **inputs** contiguously.  This check verifies [1] both the inputs and outputs are contiguous and [2] assumes there are the same number of inputs and outputs.  Some kernels (e.g. NCCL) may `Alias()` their inputs+outputs, but others may not, and some kernels may have different number of inputs/outputs.

At the least, you should separate out the input contiguous check from the output contiguous check.

Reply:
Agreed. Made it an input check only. A subsequent PR will add the options for contiguous outputs and then we can add the output check. 

---
In reply to: [505115537](https://github.com/microsoft/onnxruntime/pull/4443#discussion_r505115537) [](ancestors = 505115537)